apps/loader/loader.js

```diff
@@ -109,6 +110,37 @@ function handleUpload(file, filename) {
   );
 }
 
+function handleDownload(id) {
+  var fileName='';
+  store.getSlide(id)
+      .then((response) => {
+        return response[0]['location'];
+      }).then((location) => {
+        fileName= location.substring(8, location.length);
+        return fileName;
+      }).then((fileName) =>{
+        fetch(downloadURL + fileName, {
+          credentials: 'same-origin',
+          method: 'GET',
+        }).then((response) => response.blob())
```

Comment:
interesting consequence, the client will download the error message as a html file. I guess we have to explicitly check for an error status code?

Reply:
@birm , yeah it would be required. 
And also, I noticed then we a user tries do delete a slide which doesn't exist (try this by calling handleDelete() with an id of a slide already deleted, then this error is given in console : 
`Uncaught (in promise) TypeError: Cannot read property 'location' of undefined
    at loader.js:117`
So, I would also request making a similar change in getSlide() function, to check the error status code, as well as if response[0] has a property 'location' or not.